usecases/cluster/transactions_write.go

```diff
@@ -215,7 +320,17 @@ func (c *TxManager) IncomingCommitTransaction(ctx context.Context,
 }
 
 type Transaction struct {
-	ID      string
-	Type    TransactionType
-	Payload interface{}
+	ID       string
+	Type     TransactionType
+	Payload  interface{}
+	Deadline time.Time
+}
+
+func ContextFromTx(tx *Transaction) (context.Context, context.CancelFunc) {
```

Comment:
What is this method for?
I see it is now used only in tests, and `resetTxExpiry` has its own way of creating context

Reply:
Great catch. That part was refactored at some point. If it's only used in tests we probably won't need this anymore. Let me address this. :-)